homeassistant/components/media_player/__init__.py

```diff
@@ -848,27 +855,37 @@ async def async_browse_media(
         """
         raise NotImplementedError()
 
+    async def _async_fetch_image(self, url):
+        """Fetch image.
+
+        Images are cached in memory (the images are typically 10-100kB in size).
+        """
+        cache_images = ENTITY_IMAGE_CACHE[CACHE_IMAGES]
+        cache_maxsize = ENTITY_IMAGE_CACHE[CACHE_MAXSIZE]
 
-async def _async_fetch_image(hass, url):
-    """Fetch image.
+        if urlparse(url).hostname is None:
+            url = f"{get_url(self.hass)}{url}"
 
-    Images are cached in memory (the images are typically 10-100kB in size).
-    """
-    cache_images = ENTITY_IMAGE_CACHE[CACHE_IMAGES]
-    cache_maxsize = ENTITY_IMAGE_CACHE[CACHE_MAXSIZE]
+        if url not in cache_images:
+            cache_images[url] = {CACHE_LOCK: asyncio.Lock()}
 
-    if urlparse(url).hostname is None:
-        url = f"{get_url(hass)}{url}"
+        async with cache_images[url][CACHE_LOCK]:
+            if CACHE_CONTENT in cache_images[url]:
+                return cache_images[url][CACHE_CONTENT]
 
-    if url not in cache_images:
-        cache_images[url] = {CACHE_LOCK: asyncio.Lock()}
+        (content, content_type) = await self._async_retrieve_image(url)
 
-    async with cache_images[url][CACHE_LOCK]:
-        if CACHE_CONTENT in cache_images[url]:
-            return cache_images[url][CACHE_CONTENT]
+        async with cache_images[url][CACHE_LOCK]:
+            cache_images[url][CACHE_CONTENT] = content, content_type
+            while len(cache_images) > cache_maxsize:
+                cache_images.popitem(last=False)
 
+        return content, content_type
+
+    async def _async_retrieve_image(self, url):
```

Comment:
It's confusing to have a "retrieve" image and a "fetch" image method.

Reply:
we could do fetch_image and fetch_image_from_cache?

Reply:
that sounds better. We should also update the docs to make sure the caching one is only used for current covers, not for media browser images.

Reply:
Made this change and updated the docs.